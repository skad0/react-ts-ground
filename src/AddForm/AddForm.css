.AddForm {
    display: flex;
    flex-direction: row;
}

.AddForm-Input {

}

.AddForm-Button {

}
