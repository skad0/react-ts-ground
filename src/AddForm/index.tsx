import * as React from 'react';

import {usedRegExp} from '../helpers';

import './AddForm.css';

interface IProps {
    handleAdd: (value: string) => void;
};

class AddForm extends React.Component<IProps> {
    private input: React.RefObject<HTMLInputElement>;

    constructor(props: IProps) {
        super(props);

        this.input = React.createRef();

        this.onAdd = this.onAdd.bind(this);
    }

    public render() {
        return (
            <div className="AddForm">
                <form onSubmit={this.onAdd}>
                    <input type="text" className="AddForm-Input" ref={this.input}  />
                    <button className="AddForm-Button">+</button>
                </form>
            </div>
        );
    }

    private onAdd(event: React.FormEvent<HTMLFormElement>) {
        event.preventDefault();

        const input: HTMLInputElement|null = this.input.current;

        if (!input) {
            return;
        }

        const value: string = input.value.trim();

        if (!validateInput(value)) {
            return;
        }

        input.value = '';

        this.props.handleAdd(value);

        function validateInput(url: string):boolean {
            const matchRegexp: RegExp = usedRegExp.YOUTUBE_VIDEO_ID;

            return matchRegexp.test(url);
        }
    }
}

export default AddForm;
