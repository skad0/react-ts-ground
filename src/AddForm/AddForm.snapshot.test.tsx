import * as React from 'react';
import * as renderer from 'react-test-renderer';

import AddForm from './index';

// tslint:disable-next-line
function addHandler() {};

describe('AddForm', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<AddForm handleAdd={addHandler} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

