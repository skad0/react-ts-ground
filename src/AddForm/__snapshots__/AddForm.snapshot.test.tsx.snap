// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm renders correctly 1`] = `
<div
  className="AddForm"
>
  <form
    className="AddForm-Form"
    onSubmit={[Function]}
  >
    <input
      className="AddForm-Input"
      placeholder="Enter Youtube's url"
      type="text"
    />
    <button
      className="AddForm-Button"
    >
      +
    </button>
  </form>
</div>
`;
